### PR TITLE
kvutils: Add metadata to `Err` [KVL-1032]

### PR DESCRIPTION
**Pull request by hubert-da**
A follow-up to: 
- https://github.com/digital-asset/daml/pull/10964

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto
Comment (hubert-da) [inline ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto:605]: This is more or less what we discussed with @SamirTalwar-DA today. I think this is the smallest change to propagate more information from the `Err` without introducing more `Rejection`s. @SamirTalwar-DA, @miklos-da WDYT? Is it worth doing?

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto:605]: I think it's a good idea to introduce this first, and then break down the subclasses into their own error types in the future.

If you do it the other way around you have to do both at once.

**Comment by SamirTalwar-DA** [comment]
Why is this PR targeting another branch, rather than `main`?

Can we wait until that other branch is merged to merge this one, or are you expecting that we need to merge this one first?

File: ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto
Comment (miklos-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto:605]: Agreed, this looks like a good approach.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala
Comment (miklos-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala:584]: This will produce an output with newlines. Given that the goal of these pieces of metadata is not to print them out on the console, I'd suggest using [`shortDebugString`](https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/TextFormat.Printer.html#shortDebugString-com.google.protobuf.MessageOrBuilder-) instead, as that doesn't produce newlines and is still human readable.

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/ConversionsSpec.scala
Comment (miklos-da) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/ConversionsSpec.scala:222]: Nit: for consistency, you might want to use English expressions instead of camelCasedOnes here as well (e.g., `metadataKey` => `metadata key`).

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala
Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala]: Sorry, this is something that has been merged in the previous PR. We decided to leave this one, as it's possible to parse it with: `TextFormat.parse`, and this is what is demonstrated by a test. Are you ok with that?

File: ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto
Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto:605]: Do you have an idea about how the proto will evolve?

Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto:605]: To me, it makes sense to actually keep the `InvalidParticipantState` error, as this should be a really rare one. So, if you ask me, I don't think we're gonna change the errors soon.

**Review by SamirTalwar-DA — APPROVED** [review]

**Pull request merged**
